components/dash-core-components/src/components/Loading.react.js

```diff
@@ -164,4 +234,36 @@ Loading.propTypes = {
          */
         component_name: PropTypes.string,
     }),
+
+    /**
+     * Add a time delay (in ms) to the spinner being removed to prevent flickering.
+     */
+    delay_hide: PropTypes.number,
+
+    /**
+     * Add a time delay (in ms) to the spinner being shown after the loading_state
+     * is set to True.
+     */
+    delay_show: PropTypes.number,
+
+    /**
+     * Whether the Spinner should show on app start-up before the loading state
+     * has been determined. Default True.
+     */
+    show_initially: PropTypes.bool,
+
+    /**
+     * Specify component and prop to trigger showing the loading spinner
+     * example: `[{"output-container": "children"}]`
```

Comment:
Does this mean 'while pending'? e.g. while server-side callback is executing?

Reply:
Yes - the `dcc.Loading` component can wrap the whole layout - but maybe you only want the loading spinner to show when only certain components are being updates.  I'll add some examples here shortly

Reply:
Oh I see. I'm going to suggest a comment then above.